sdks/python/apache_beam/io/tfrecordio.py

```diff
@@ -78,7 +78,10 @@ def _masked_crc32c(cls, value, crc32c_fn=_default_crc32c_fn):
       Masked crc32c checksum.
     """
 
-    crc = crc32c_fn(value)
+    if isinstance(value, bytes):
```

Comment:
Would anything break if we always follow the `else` branch? If yes, why? 
Also, last letter in `crc32c_fu` is a typo?

Reply:
Before this change, the default codepath was:
`crc = crc32c_fn(value)`
I assume this did not work on Python 3, so, I suspected that on Python 3 `value` is a `str`. Are you saying that `value` is sometimes `str` and sometimes `bytes` on Python 3? If yes, why is value not always the same type?

Reply:
+1. I want us to have clarity on which types we expect throughout methods of TFRecord IO, we should avoid having  APIs that accept both `str=unicode` and `bytes`. See: https://docs.python.org/3/howto/pyporting.html#text-versus-binary-data. If it makes sense for TF Records to hold unencoded strings, then we may want to encode them to bytes right before writing them, and decode them as soon as we read them.

Reply:
Thanks for all the reviews 😃 I rolled back my previous change and only make sure the write_record function always operate on bytes for both py2 and py3